### ISSUE TITLE
Hugo 0.16-DEV (Jan 2016) fails to sync theme's static directory on Linux

### ISSUE THREAD

**Issue by anthonyfok** [spf13/hugo#1802]
@pcdummy first reported yesterday in Gitter:

> Having troubles with hugo: it's not syncing the static contents from the theme. Using the latest HEAD hugo.
> what OS and file system?  Linux and BTRFS.  Ubuntu 15.10 MATE edition.

I was soon able to reproduce the same on Debian GNU/Linux (sid), amd64.

@Zenithar also reported that assets are not synced on Linux:

```
INFO: 2016/01/30 17:37:53 hugo.go:393: Using config file: /home/tnormand/Repositories/github/zenithar/zenithar.hugo/config.toml
INFO: 2016/01/30 17:37:53 hugo.go:483: using a UnionFS for static directory comprised of:
INFO: 2016/01/30 17:37:53 hugo.go:484: Base: /home/tnormand/Repositories/github/zenithar/zenithar.hugo/themes/bleak/static
INFO: 2016/01/30 17:37:53 hugo.go:485: Overlay: /home/tnormand/Repositories/github/zenithar/zenithar.hugo/static/
INFO: 2016/01/30 17:37:53 hugo.go:517: syncing static files to /
INFO: 2016/01/30 17:37:53 site.go:1274: found taxonomies: map[string]string{"tag":"tags", "topic":"topics"}
INFO: 2016/01/30 17:37:53 htmlredirect.go:116: Alias "/tags/pssh/page/1" translated to "tags/pssh/page/1/index.html"
```

**Comment by anthonyfok** [comment]
Some more notes:

1. @spf13 could not reproduce this bug on OS X, so it is probably Linux-specific.
2. It may have to do with the new UnionFS feature in Afero.

See discussion on https://gitter.im/spf13/hugo starting with the first comment by @pcdummy.

**Comment by hrq** [comment]
I have same problem on osx10.11.3.

~~~
INFO: 2016/01/31 02:57:19 hugo.go:393: Using config file: /Users/hrq/Projects/blog-hugo/config.toml
INFO: 2016/01/31 02:57:19 hugo.go:483: using a UnionFS for static directory comprised of:
INFO: 2016/01/31 02:57:19 hugo.go:484: Base: /Users/hrq/Projects/blog-hugo/themes/twentysixteen/static
INFO: 2016/01/31 02:57:19 hugo.go:485: Overlay: /Users/hrq/Projects/blog-hugo/static/
INFO: 2016/01/31 02:57:19 hugo.go:517: syncing static files to /Users/hrq/Projects/blog-hugo/src/public/
~~~

The message is ok. But not sync theme's static dir.

~~~
afero.NewCopyOnWriteFs(base, overlay)
~~~

It seems not working.

**Comment by leonard7e** [comment]
Hi,

I am running into same problems, since I upgraded to hugo version v0.16-DEV. 
Hugo running on Ubuntu Wily Werewolf here.

Static files of a theme I use are not synced. 
However, if I generate html directly (without starting server) my site is generated correctly.

**Comment by tatsushid** [comment]
According to the `git bisect` result, this problem seems to be introduced from 7e196a8. I'm testing on MacOS X 10.9.5.

**Comment by tatsushid** [comment]
OK, I found the root cause of the problem. Afero [`CopyOnWriteFS.Stat` function](https://github.com/spf13/afero/blob/master/copyOnWriteFs.go#L76) checks if specified name file or directory exists on overlay file system and if this check failed with `syscall.ENOENT`, it tries to check base file system with the same name.

But if the original file system of overlay is `os`, it returns `os.PathError`, not `syscall.ENOENT` so in this case, it silently ignores base file system one.

I will send a pull request to afero to fix this problem.

**Comment by anthonyfok** [comment]
@tatsushid You are really awesome!
Thank you for investigating and fixing the problem!

**Comment by Zenithar** [comment]
@tatsushid  Very good ^^

**Comment by tatsushid** [comment]
This should be fixed in spf13/afero@4c1d2aa. Please update afero and try.

**Comment by anthonyfok** [comment]
Indeed it is!  With the updated spf13/afero@4c1d2aa, both `hugo` and `hugo server` works perfectly with the theme statics directory properly synchronized.

Many thanks @tatsushid!  You are the best!  ありがとうございました！
